Source/monster.cpp

```diff
@@ -4503,20 +4502,20 @@ void MissToMonst(Missile &missile, Point position)
 	}
 }
 
-Monster *MonsterAtPosition(Point position)
+Monster *MonsterAtPosition(Point position, bool strict)

```

Comment:
Rather than making it a flag, let's have 2 functions with different return types:

```c++
// Assumes `position` is in dungeon bounds and contains a monster.
Monster &MonsterAtPosition(Point);

// Returns the monster if the `position` is in dungeon bounds and contains a monster,
// `nullptr` otherwise.
Monster *MaybeMonsterAtPosition(Point);
```

Not sure about the names either but having 2 functions does seem sensible.

Reply:
The "strict" version can also return a `nullptr`. The only different is that it won't check for walking monsters

Reply:
With the way these functions are currently used I personally think that `ObjectAtPosition`, `PlayerAtPosition`, and `MonsterAtPosition` are more natural returning pointers. The position based lookups are similar to `map::find()` which tend to return `iterator::end` when the find fails, though the name does imply that it's equivalent to `map::at()`...

Putting the logic to filter moving players/monsters and large objects into `Is*AtPosition` (as an equivalent to `map::contains()` and making the at method just do a simple lookup could be worth it though.

Reply:
I am a bit confused about what I should do here 😄 

This PR is not changing the return type. It's still returning pointers and not references. The only thing that the extra does is to indicate if `MonsterAtPosition` should ignore walking monsters or not.

I could also have it as a separate function instead of a flag, but then naming becomes a challenge. Perhaps `StandingMonsterAtPosition`?

Reply:
Nah, keeping the one function for now is fine imo. `MonsterAtPosition` is barely used at the moment, need to expand the usage and work out if splitting it like `ObjectAtPosition`/`FindObjectAtPosition` makes sense.

If you want you could rename `MonsterAtPosition` to `FindMonsterAtPosition` (keep the param you've added, just rename all current uses). We can then add a version returning a reference if it's useful.

Reply:
Alright! Function renamed